File: 217.存在重复元素.go
/*
 * @lc app=leetcode.cn id=217 lang=golang
 *
 * [217] 存在重复元素
 */

package main

// @lc code=start
func containsDuplicate(nums []int) bool {
	count := make(map[int]int, 0)

	for _, num := range nums {
		count[num] += 1
		if count[num] >= 2 {
			return true
		}
	}
	return false
}

// @lc code=end

File: 217.存在重复元素.rs
/*
 * @lc app=leetcode.cn id=217 lang=rust
 *
 * [217] 存在重复元素
 */

use crate::Solution;

// @lc code=start
impl Solution {
    pub fn contains_duplicate(mut nums: Vec<i32>) -> bool {
        nums.sort();
        for i in 0..nums.len().checked_sub(1).unwrap_or(0) {
            if nums[i] == nums[i + 1] {
                return true;
            }
        }
        false
    }
}
// @lc code=end

